### ISSUE TITLE
Initialisation on Android 8.0.0.

### ISSUE THREAD

**Issue by MicPra** [kaaholst/android-squeezer#685]
I know it's an old Android version but when I start Squeezer on my old S7 in 9 from 10 times it just shows the hourglass forever. Tap 'Playing Now' and 'Home'  - everything is working....

This is reproduceable on 2nd Android 8.0.0. but not on Android 11.0.0. As far as I rembember with 2.0.x I never had this issue.

![Screenshot_20210328-212248_Squeezer](https://user-images.githubusercontent.com/81591642/112898874-b3972300-90e1-11eb-9e22-90f20aed5829.jpg)

**Comment by kaaholst** [comment]
I've seen this happening occasionally, there appears to be a race condition for the home menu.

Android 8.0.0 is indeed supported, so I'll try to fix this.

In the meantime, I think you have found the proper workaround.

**Comment by bvrulez** [comment]
When it works the log is: 

```
ItemListActivity: ActivePlayerChanged: Player{...}
ItemListActivity: showLoading: 
ItemListActivity: maybeOrderPage: 
ItemListActivity: onItemsReceived(10, 0, 10)
ItemListActivity: showContent: 

```

Then a handshake happens, is completed (several times) but then it never calls showContent:

```
I/HttpStreamingTransport: /meta/handshake advice: {reconnect=retry, interval=0, timeout=60000}
I/CometClient: Connected, start learning server capabilities
...
I/ConnectionState: Handshake complete: HandshakeComplete{version='8.1.1'}

```

**Comment by bvrulez** [comment]
While "loading" the `JSON is not valid`. In this case the `unprocessed` variable always has a length of around 12.000.  Every time the problem gets solved it will just add one more (small) chunk of about 1.000 bytes and the JSON is valid in one step.

**Comment by bvrulez** [comment]
`isValidJson(unprocessed)` returns false due to a JSONException "Unterminated string". 

The loop should then start again with `while(!"0".equals(readLine()))` but the line returns null. 

I am unsure if the loop works as expected. If `readLine()` has to be done to check if the loop should be executed, is there not this input missing later inside the loop, when the next `readLine()` gets the content to be analysed?

**Comment by bvrulez** [comment]
I now understand that the reading is line by line and highly dependend upon the order of the content coming from the server. 

`while (!"0".equals(line = readLine()))` is never not true because no line with "0" is ever sent.

**Comment by kaaholst** [comment]
This fixed from version 2.3.0, so closing this.

Thank you!
